internal/internal_utils.go

```diff
@@ -110,8 +121,10 @@ func newChannelContext(ctx context.Context, options ...func(builder *contextBuil
 			// Make sure to not set rpc timeout lower than minRPCTimeout
 			if rpcTimeout < minRPCTimeout {
 				rpcTimeout = minRPCTimeout
-			} else if rpcTimeout > maxRPCTimeout {
+			} else if rpcTimeout > maxRPCTimeout && !isQuery {
 				rpcTimeout = maxRPCTimeout
+			} else if rpcTimeout > maxQueryRPCTimeout && isQuery {
+				rpcTimeout = maxQueryRPCTimeout
```

Comment:
If its easy to add, can we have a unit test which checks:
if not query, rpcTimeout = 5 seconds
if query, rpcTimeout = 20 seconds